### ISSUE TITLE
[Feature] Controlling Content Elements' HTML Properties

### ISSUE THREAD

**Issue by phonique** [TryGhost/Ghost#4846]
This is a feature request for the ghost-ui in response to [halfdan's](https://ghost.org/forum/plugins/15941-custom-post-content-adding-css-classes-onto-sections-in-blog-posts/) forum request, but more generalized.

I'd like to suggest adding a small icon next to each tag in the preview panel with an (extensible) options drop-down/modal to influence relevant/available html attributes of that tag (e.g. the class and id of all tags, target for links, etc.). Another icon on top, or next to the gear at the bottom could enable/disable this functionality, similar to the [pilcrow](https://en.wikipedia.org/wiki/Pilcrow) that enables its display in word processing.

The motivation here is primarily in keeping the ability to add image placeholders (and the inherent ability to drag&drop them into the preview panel), while enabling a greater degree of customization. 
An example application would be to set certain post images to a specific class to be used in a (e.g. lightbox) gallery, while leaving others out.

As we already have interaction in the preview pane (i.e. image drag-in and the image-delete button), this would fit nicely in the existing paradigm.

**Comment by halfdan** [comment]
I am flattered but I think you meant to reference @ErisDS :)

**Comment by ErisDS** [comment]
Hi @phonique, I'm sorry, I'm really not sure at all what you're explaining here.

My comment on the forum was intended to explain that the editor we have right now supports full HTML or markdown, but doesn't have extension points. 

I'm really not sure what this in relation to, or what problem it's trying to solve. Tags have their own management user interface which allows you to add post images to them, and if you want to add specific classes to things, that really belongs in the theme. 

Without specific use cases I'm not really able to move this forward. I'm going to close it for now. By all means provide an update to clarify and we can reopen it.

**Comment by phonique** [comment]
Hi Eris,
with tags I mean the HTML tags, of course, for which markdown is just a shortcut.

As there's no "traditional" rich text editor (i.e. one, where a user could switch between the source and WYSIWIG views), but markdown is provided as an abstraction instead, the suggestion/feture would allow attaching specific CSS classes to parts of an article (e.g. a paragraph, heading or image), which would otherwise be only possible to do via HTML - so, not possible for the majority of assumed end users. (Ghost is aiming for user-friendliness, right?)

An expansion of this feature could provide hooks to pre-define the possible choices of classes, e.g. to have a series of images belong to one gallery, while others to be displayed regularly.

These are things that are impossible to do via theming, unless posted articles always follow the same structure (e.g. N pictures after the first heading - class foo, rest no class.)

I hope I've made the use case clearer.

In other words this feature allows to remove the necessity to switch to HTML when declaring certain parts/elements of an article as a certain CSS class/ID - something that is useful for the non-technical crowd (and would possibly allow the theme devs to pre-define specific css-classes for easy selection by the end user).
